### PR TITLE
Give appropriate labels to GraphScope in Kubernetes

### LINKED ISSUE #938: Give appropriate labels to graphscope when deployed in kubernetes.
Assign appropriate labels to pods/services/deployments of graphscope can make it much easier to delete deployed graphscope, as labels can group components in one cluster together. or we need to delete those components one by one (When deployed by Python) like:
```shell
# Suppose the name is xxx
kubectl delete deployments gs-engine-xxx
kubectl delete pods gs-etcd-xxx
```
Instead it's better to use labels to manage them, like:
```shell
kubectl delete pods -l app.kubernetes.io/instance=xxx
```
 And it's more aligned with the best practice, make room for future management, rollbacks, updates, scale, etc.

And we can add this feature to Helm charts as well.
- [ ] Deployed by Python
- [x] Deployed by Helm 

Reference: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

### PR DESCRIPTION
**Pull request by lidongze0629**
## Related issue number

Fixes #938

### CODE REVIEW

File: coordinator/gscoordinator/cluster.py [hunk 1]
```diff
@@ -418,7 +419,27 @@ def close_interactive_instance(self, object_id):
 
     def _create_mars_scheduler(self):
         logger.info("Launching mars scheduler pod for GraphScope ...")
-        labels = {"name": self._mars_scheduler_name}
+
+        labels = {
+            "app.kubernetes.io/name": self._mars_scheduler_name,
+            "app.kubernetes.io/instance": self._instance_id,
+            "graphscope.components": "mars",
```
Comment (yecol) [inline coordinator/gscoordinator/cluster.py:426]: it is not appropriate to include mars as graphscope.components.

File: coordinator/gscoordinator/cluster.py [hunk 2]
```diff
@@ -418,7 +419,27 @@ def close_interactive_instance(self, object_id):
 
     def _create_mars_scheduler(self):
         logger.info("Launching mars scheduler pod for GraphScope ...")
-        labels = {"name": self._mars_scheduler_name}
+
+        labels = {
+            "app.kubernetes.io/name": self._mars_scheduler_name,
+            "app.kubernetes.io/instance": self._instance_id,
+            "graphscope.externals": "mars",
```
Comment (sighingnow) [inline coordinator/gscoordinator/cluster.py:426]: `. externals` ?

Comment (yecol) [reply coordinator/gscoordinator/cluster.py:426]: @sighingnow give a suggestion?

Comment (sighingnow) [reply coordinator/gscoordinator/cluster.py:426]: How about `.component`?

**Review by siyuan0322 — COMMENTED** [review]
> app.kubernetes.io/name | The name of the application | mysql | string

I suggest the `app.kubernetes.io/name` always equal to `graphscope`.
Maybe also add a `version` field?
And we use `instance` to distinguish difference graphscope clusters.
I think the `engine_name` doesn't need to appear in labels.

Reference: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

**Comment by lidongze0629** [comment]
> > app.kubernetes.io/name | The name of the application | mysql | string
> 
> I suggest the `app.kubernetes.io/name` always equal to `graphscope`. Maybe also add a `version` field? And we use `instance` to distinguish difference graphscope clusters. I think the `engine_name` doesn't need to appear in labels.
> 
> Reference: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

Thanks for your recommended labels. I will update labels to
```
app.kubernetes.io/name: graphscope
app.kubernetes.io/instance: unique_instance_id
app.kubernetes.io/version: "0.8.0"
app.kubernetes.io/component: engine/coordinator/etcd
app.kubernetes.io/component-name: engine_name/coordinator_name/etcd_name
```
`engine_name` is must for service selector

**Comment by siyuan0322** [comment]
> unique_instance_id

What about using the label `instance` together with `component`, which can select the proper pods and you won't need to update the engine name. This can be useful for our future scaling scenario, which can automatically include the newly added pod in.

**Comment by lidongze0629** [comment]
> What about using the label `instance` together with `component`, which can select the proper pods and you won't need to update the engine name. This can be useful for our future scaling scenario, which can automatically include the newly added pod in.

I think the labels above is enough for selector.

**Comment by siyuan0322** [comment]
> > What about using the label `instance` together with `component`, which can select the proper pods and you won't need to update the engine name. This can be useful for our future scaling scenario, which can automatically include the newly added pod in.
> 
> I think the labels above is enough for selector.

OK~ I was replied to the `engine-name` part, which is now gone.

**Comment by sighingnow** [comment]
> I suggest the `app.kubernetes.io/name` always equal to `graphscope`.

+1

**Review by sighingnow — APPROVED** [review]

**Pull request merged**